website/docs/language/state/encryption.mdx

```diff
@@ -0,0 +1,115 @@
+
+You will also need to decide what kind of key you would like to use. You can either opt for a static passphrase which you manually rotate regularly, or you can choose a key management system. If feasible, we recommend using a key management system that can rotate keys automatically to give you the maximum amount of protection.
+
+If you decide for a key management system, please make sure to configure regular key rotation. This is especially important if the chosen encryption method can exhaust the key. (This is not a problem with passphrase-based static keys.) You can find more information in the [encryption methods](#encryption-methods) section below.
+
+Finally, before enable encryption, please exercise your disaster recovery plan and make a temporary backup of your unencrypted state file. Once you enable encryption, OpenTofu cannot read your state file without the correct key.
+
+## Configuration
+
+You can configure encryption in OpenTofu either by specifying the configuration in the OpenTofu code, or using the `TF_ENCRYPTION` environment variable. Both solutions are equivalent and if you use both, OpenTofu will merge the two configurations, overriding any code-based settings with the environment ones.
+
+The basic configuration structure looks as follows:
+
+<Tabs>
+    <TabItem value="code" label="Code" default>
+        <CodeBlock language={"hcl"}>{ConfigurationTF}</CodeBlock>
+    </TabItem>
+    <TabItem value="env-sh" label="Environment (Linux/UNIX shell)">
+        <CodeBlock language={"shell"}>{ConfigurationSH}</CodeBlock>
+    </TabItem>
+    <TabItem value="env-ps1" label="Environment (Powershell)">
+        <CodeBlock language={"powershell"}>{ConfigurationPS1}</CodeBlock>
+    </TabItem>
+</Tabs>
+
+:::warning
+
+Once your data is encrypted, do not rename key providers and methods in your configuration! The encrypted data contains metadata related to their specific names. Instead, use a [fallback block](#key-and-method-rollover).
+
+:::
+
+:::tip
+
+You can use the [JSON configuration syntax](/docs/language/syntax/json/) instead of HCL for encryption configuration.
+
+:::
+
+:::tip
+
+If you use environment configuration, you can include the following code configuration to prevent unencrypted data from being written in the absence of an environment variable:
+
+<CodeBlock language="hcl">{Enforce}</CodeBlock>
+
+:::
+
+## Key providers
+
+There are currently no supported key providers.
+
+## Methods
+
+### AES-GCM
+
+The only currently supported encryption method is AES-GCM. You can configure it in the following way:
+
+<CodeBlock language="hcl">{AESGCM}</CodeBlock>
+
+:::warning
+
+When using a key-management system as a key provider, always configure automatic key rotation in conjunction with AES-GCM, otherwise your security may be compromised! (This does not affect passphrase-based local keys.)

```

Comment:
Frankly, with our official implementations and no plugins, I don't think this will affect **any** key providers.

Reply:
Since they will be generating a new data encryption key every time.

Reply:
```suggestion
AES-GCM is a secure, industry-standard encryption algorithm, but suffers from "key saturation". In order to configure a secure setup, you should either use a key-derivation key provider (such as PBKDF2) with a long and complex passphrase, or use a key management system that automatically rotates keys regularly. Using short, static keys will degrade your encryption.
```